doozer/doozerlib/backend/konflux_image_builder.py

```diff
@@ -184,6 +184,16 @@ async def build(self, metadata: ImageMetadata):
                 succeeded_condition = artlib_util.KubeCondition.find_condition(pipelinerun, 'Succeeded')
                 outcome = KonfluxBuildOutcome.extract_from_pipelinerun_succeeded_condition(succeeded_condition)
 
+                # Even if the build succeeded, if the SLSA attestation cannot be retrieved, it is unreleasable.
+                if outcome is KonfluxBuildOutcome.SUCCESS:
+                    image_pullspec = next((r['value'] for r in pipelinerun.status.results if r['name'] == 'IMAGE_URL'), None)
+                    try:
+                        # Get SLA attestation from konflux. The command will error out if it cannot find it.
+                        await artlib_util.get_konflux_slsa_attestation(image_pullspec, os.environ['KONFLUX_ART_IMAGES_USERNAME'], os.environ['KONFLUX_ART_IMAGES_PASSWORD'])
+                    except Exception as e:
```

Comment:
Hm I don't think it's a good practice to rely on a generic exception being raised. Instead if you have an http error code or message which indicates that the attestation is missing and then return None from the func, that will be a better indicator

Reply:
ah since it's a cmd, then rc != 0 indicates that there was a failure, which should be sufficient

Reply:
`cmd_gather_async` inside `get_konflux_slsa_attestation`  will raise a `ChildProcessError` if the command fails. So I can update `Exception` to that, but does it matter here since we are logging the error and explaining the reason?

Reply:
Not if you have check=False, then you can rely on rc

Reply:
exceptions can be raised for many reasons, for example if there is a command syntax error or even a python error in method. so imo we should look for the exact fail reason (as exact as possible)

Reply:
ack - updated in https://github.com/openshift-eng/art-tools/pull/1570/commits/0d3670c9dd295b604d2ac718e698dcf95f0b59ad